src/matrix_utilities.jl

```diff
@@ -6,23 +6,72 @@ function LinearAlgebra.logdet(H::Hermitian)
     return mag + log(resign)
 end
 
+"""
+    qr2chol(pre_array::AbstractMatrix)
```

Comment:
`qr2chol` strongly suggests to me that it is given a QR decomposition and returns a Cholesky decomposition.
But what this seems to be doing is taking in a square-root of a matrix, and making it into a valid cholesky factor.

Maybe: `choleskify`? Or even just `triangularize`? To me the latter feels quite appropriate because that's 90% of what's happening, but it ignores the whole "making the diagonal positive" part so idk.

Reply:
It seems to me `choleskify` carries the same problems as present naming, `triangularize` seems a bit vague.
While it doesnt return a cholesky, it does return a valid cholesky factor I would be okay with `qr2cholf` or something more verbose. Though on the other hand, what is happening is that it is converting an arbitrary square-root factor `X'*X` into a valid Cholesky factor `U'*U`. 

As for `pre_array`, I believe this is standard nomenclature in square-root filtering literature so I would keep it.

Reply:
I settled on  `rsqrt2cholU`. As the function takes a right square-root and converts it into a right Cholesky factor (U) this should be descriptive enough.